django/forms/utils.py

```diff
@@ -158,10 +158,14 @@ def from_current_timezone(value):
     When time zone support is enabled, convert naive datetimes
     entered in the current time zone to aware datetimes.
     """
+    return_value = value
     if settings.USE_TZ and value is not None and timezone.is_naive(value):
         current_timezone = timezone.get_current_timezone()
         try:
-            return timezone.make_aware(value, current_timezone)
+            return_value = timezone.make_aware(value, current_timezone)
+            if not (timezone._is_pytz_zone(current_timezone) or
+                    not timezone._datetime_ambiguous_or_imaginary(value, current_timezone)):
+                raise ValueError('Ambiguous or non-existent time.')
```

Comment:
Do we have test for this error?

Reply:
Yes, this is covered in `tests/timezones/tests.py` `LegacyFormsTests.test_form_with_ambiguous_time()`